graphwalker-io/src/test/java/org/graphwalker/io/factory/yed/YEdContextFactoryTest.java

```diff
@@ -340,4 +343,241 @@ public void createfromString() throws IOException {
+     * edge: 		n
+     * default: 	n
+     * overRide: 	n
+     * desc:		node: y edge: y
+     */
+    Context ctx = getContextFromGraphml("graphml/customProp/Issue25_Graph05.graphml");
+    RuntimeModel m = ctx.getModel();
+
+    for (RuntimeVertex v : m.getVertices()){
+      assertGeometryPropertiesAreOk(v);
+      Assert.assertTrue("Node description is missing.", v.hasProperty("description"));
+      Assert.assertEquals("Node Description", v.getProperty("description").toString());
+      Assert.assertFalse("CustomNodeProperty is erroneously on Node.", v.hasProperty("CustomNodeProperty"));
+      Assert.assertFalse("CustomEdgeProperty is erroneously on Node.", v.hasProperty("CustomEdgeProperty"));
+    }
+
+    for (RuntimeEdge e : m.getEdges()){
+      Assert.assertTrue("Edge description is missing.", e.hasProperty("description"));
+      Assert.assertEquals("Edge Description", e.getProperty("description").toString());
+      Assert.assertFalse("CustomEdgeProperty is erroneously on  Edge.", e.hasProperty("CustomEdgeProperty"));
+      Assert.assertFalse("CustomNodeProperty is erroneously on Edge.", e.hasProperty("CustomNodeProperty"));
+    }
+  }
+
+  @Test
+  public void customProperties_06() throws IOException{
+    /*
+     * node: 		n
+     * edge: 		y
+     * default: 	y
+     * overRide: 	y
+     * desc:		node: n edge: n
+     */
+    Context ctx = getContextFromGraphml("graphml/customProp/Issue25_Graph06.graphml");
+    RuntimeModel m = ctx.getModel();
+
+    for (RuntimeVertex v : m.getVertices()){
+      assertGeometryPropertiesAreOk(v);
+      Assert.assertFalse("Description is erroneously on Node.", v.hasProperty("description"));
+      Assert.assertFalse("CustomEdgeProperty is erroneously on Node.", v.hasProperty("CustomEdgeProperty"));
+    }
+
+    for (RuntimeEdge e : m.getEdges()){
+      Assert.assertTrue("CustomEdgeProperty is missing from Edge.", e.hasProperty("CustomEdgeProperty"));
+      Assert.assertEquals("OverRideEdgeValue", e.getProperty("CustomEdgeProperty").toString());
+      Assert.assertFalse("Description is erroneously on Edge.", e.hasProperty("description"));
+    }
+  }
+
+  @Test
+  public void customProperties_07() throws IOException{
+    /*
+     * Graph with a subgraph. Nodes and edges with 3 custom properties each.
+     * Interestingly the node default values appear in yED but the <data key=xx> does not appear in the .graphml
+     *
+     * node: 		y
+     * edge: 		y
+     * default: 	y CNP 1 & CNP 3; CEP 1 & CEP 3
+     * overRide: 	y CNP 2; CEP 2
+     * desc:		y
+     * edge:      y
+     */
+    Context ctx = getContextFromGraphml("graphml/customProp/Issue25_Graph07.graphml");
+    RuntimeModel m = ctx.getModel();
+
+    List<RuntimeVertex> vertices = m.getVertices();
+    Assert.assertThat(vertices.size(),is(3));
+
+    for (RuntimeVertex v : vertices){
+      assertGeometryPropertiesAreOk(v);
+      Assert.assertTrue("CustomNodeProperty 1 is missing from Node.", v.hasProperty("CNP 1"));
```

Comment:
You can use static import for Assert (don't think we use it ourself everywhere but I have thought about changing that)